app/src/main/java/com/github/se/wanderpals/ui/screens/dashboard/DashboardFinanceWidget.kt

```diff
@@ -212,3 +213,39 @@ fun ExpenseItem(expense: Expense, currencyCode: String) {
             modifier = Modifier.padding(8.dp).testTag("expenseAmount" + expense.expenseId))
       }
 }
+
+/**
+ * Function to format the total expense amount to a more readable format. The function takes the
+ * total expense amount as a Double and returns a formatted String. The function formats the amount
+ * to a maximum of 2 decimal places and adds a suffix of M, B, or T based on the number of digits in
+ * the amount.
+ *
+ * @param totalexpense The total expense amount as a Double.
+ */
+private fun formatExpense(totalexpense: Double): String {
+  var formattedString = String.format(Locale.US, "%.02f", totalexpense).reversed()
+
+  val numString = formattedString.substring(3)
+  val numberOfDigits = numString.length / 3
+  if (numberOfDigits > 1) {
+    val formattedReversedString = numString.chunked(3).joinToString(".")
+
+    // take only the digits after the last point and the two digits before the point
+    val expenseToString = formattedReversedString.substringAfterLast(".").substringBeforeLast(".")
+
+    formattedString =
+        when (numberOfDigits) {
+          2 -> (expenseToString.reversed() + "M")
+          3 -> (expenseToString.reversed() + "B")
+          4 -> (expenseToString.reversed() + "T")
+          else -> {
+            "Error"
+          }
+        }
+  } else {
+    formattedString = formattedString.reversed()
+    Log.d("formattedString", formattedString)
```

Comment:
The Log.d statement in the formatExpense function is useful for debugging. Should it be left in the final version?

Reply:
Sure, i have to remove it